src/validations/html/rules/MismatchedTag.js

```diff
@@ -0,0 +1,139 @@
+import {List, Map, Record, Set} from 'immutable';
```

Comment:
However, I converted `this._openTagStack` from `List` to a native `Array` since `Array` performs naturally well for a stack, the `List` appears to have no advantages.

Meanwhile, `this._unopenedTagQueuesByName` represent queues, and `List` [promises](https://facebook.github.io/immutable-js/docs/#/List) "efficient addition and removal from both the end (`push`, `pop`) and beginning (`unshift`, `shift`)." Therefore, I've kept `List` for the queues.